File: frontend/static/scripts/Task.js
export class Task {
    constructor(id, name, text, difficulty, start, end, domElement, completed = false) {
        this.id = id;
        this.name = name;
        this.text = text;
        this.start = start;
        this.end = end;
        this.difficulty = difficulty;
        this.domElement = domElement;
        this.completed = completed;
    }

    updateFields(newTask){
        this.name = newTask.name;
        this.text = newTask.text;
        this.start = newTask.start;
        this.end = newTask.end;
        this.difficulty = newTask.difficulty;
        this.domElement = newTask.domElement;
    }
}

File: models/task.py
import datetime
from typing import Annotated

from pydantic import BaseModel
from sqlmodel import SQLModel, Field


class Task(SQLModel, table=True):
    id: Annotated[int, Field(primary_key=True)]
    user_id: int
    name: str
    text: str
    start: datetime.date
    end: datetime.date
    difficulty: int


class TaskAdd(BaseModel):
    name: str
    text: str
    start: datetime.datetime
    end: datetime.datetime
    difficulty: int


class TaskUpdate(BaseModel):
    id: int
    name: str
    text: str
    start: datetime.datetime
    end: datetime.datetime
    difficulty: int
